packages/amplify-provider-awscloudformation/src/pre-push-cfn-processor/cfn-pre-processor.ts

```diff
@@ -24,3 +39,176 @@ export async function preProcessCFNTemplate(filePath: string): Promise<string> {
+) {
+  if (!(service === 'Lambda' || service === 'ElasticContainer')) {
+    return;
+  }
+  const { templateFormat, cfnTemplate } = await readCFNTemplate(path.join(resourceDir, cfnFile));
+  const customPolicies = stateManager.getCustomPolicies(category, resourceName);
+  if (!validateExistCustomPolicies(customPolicies)) {
+    if (cfnTemplate.Resources.CustomLambdaExecutionPolicy) {
+      delete cfnTemplate.Resources.CustomLambdaExecutionPolicy;
+      await writeCFNTemplate(cfnTemplate, filePath, { templateFormat });
+    }
+    if (cfnTemplate.Resources.CustomExecutionPolicyForContainer) {
+      delete cfnTemplate.Resources.CustomExecutionPolicyForContainer;
+      await writeCFNTemplate(cfnTemplate, filePath, { templateFormat });
+    }
+    return;
+  }
+  await validateCustomPoliciesSchema(customPolicies, category, resourceName);
+
+  await addCustomPoliciesToCFNTemplate(service, category, customPolicies, cfnTemplate, filePath, resourceName, {templateFormat} );
+
+}
+
+//merge the custom IAM polciies to CFN template for lambda and API container
+
+async function addCustomPoliciesToCFNTemplate(
+  service: string,
+  category: string,
+  customPolicies: CustomIAMPolicies,
+  cfnTemplate: Template,
+  filePath: string,
+  resourceName: string,
+  {templateFormat}: any
+  ) {
+  let customExecutionPolicy;
+  const { envName } = stateManager.getLocalEnvInfo();
+  if(service === 'Lambda') {
+    customExecutionPolicy = customExecutionPolicyForFunction;
+  }
+  if (service === 'ElasticContainer') {
+    const roleName = cfnTemplate.Resources.TaskDefinition.Properties.ExecutionRoleArn["Fn::GetAtt"][0];
+    customExecutionPolicy = customExecutionPolicyForContainer;
+    const role = Fn.Ref(roleName);
+    customExecutionPolicy.Properties.Roles.push(role);
+  }
+
+  warnWildcardCustomPolicies(customPolicies, resourceName);
+
+  for (const customPolicy of customPolicies) {
+    validateCustomPolicy(customPolicy, category, resourceName);
+    const policyWithEnv = await replaceEnvForCustomPolicies(customPolicy, envName);
+    if (!policyWithEnv.Effect) policyWithEnv.Effect = iam.Effect.ALLOW;
+    customExecutionPolicy.Properties.PolicyDocument.Statement.push(policyWithEnv);
+  }
+
+  if (service === 'Lambda') {
+    cfnTemplate.Resources.CustomLambdaExecutionPolicy = customExecutionPolicy;
+  }
+  if (service === 'ElasticContainer') {
+    cfnTemplate.Resources.CustomExecutionPolicyForContainer = customExecutionPolicy;
+  }
+
+  await writeCFNTemplate(cfnTemplate, filePath, { templateFormat });
+}
+
+
+//validate the format of actions and ARNs for custom IAM policies
+function validateCustomPolicy(customPolicy: CustomIAMPolicy, category: string, resourceName: string) {
+  const resources = customPolicy.Resource;
+  const actions = customPolicy.Action;
+  const resourceRegex = new RegExp('arn:(aws[a-zA-Z0-9-]*):([a-zA-Z0-9\\-])+:([a-z]{2}(-gov)?-[a-z]+-\\d{1})?:(\\d{12})?:(.*)');
+  const actionRegex = new RegExp('[a-zA-Z0-9]+:[a-z|A-Z|0-9|*]+');
```

Comment:
you may be able to add these regexs to the json schema and get this validation for free with ajv
http://json-schema.org/understanding-json-schema/reference/regular_expressions.html

Reply:
I cannot find a way to print out the wrong resource or action using ajv to validate for regex

Reply:
```
if(!validatePolicy(data)) {
    let errorMessage = ``;
    validatePolicy
      .errors
      .forEach(error => error.keyword === 'pattern'?
        errorMessage += `Invalid custom IAM policy for ${resourceName}. Incorrect "Resource": ${error.data}\n Edit ${customPoliciesPath} to fix`
        :errorMessage += `Invalid custom IAM policies in the ${resourceName} ${categoryName}.\n
        Edit <project-dir>/amplify/backend/function/${resourceName}/custom-policies.json to fix
        Learn more about custom IAM policies for ${categoryName}: https://docs.amplify.aws/function/custom-policies\n)`;
    throw new CustomPoliciesFormatError(errorMessage);
  }
```
Above is what I tried but didn't work. The error.data is undefined